### PR TITLE
HPCC-15345 Swap change in init_thor

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner Michael.Gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@jakesmith Moved the swap section to only run on TEC_Swap return.

**Comment by jakesmith** [comment]
Not sure about this. Looking back, it used to run swapnode unconditionally, except when thor exited cleanly. Which I believe it would have done on a idle restart (the TEC_Idle return code was new).

Other unexpected states were intended to run swapnode, not only when thormaster specifically signal that needed.

So this should be changed to avoid swapnode if TEC_Idle, but otherwise proceed.

**Comment by jakesmith** [comment]
@Michael-Gardner back to you

**Comment by Michael-Gardner** [comment]
@jakesmith I swapped the logic for TEC_Swap and TEC_Idle.  Now everything should have the same behavior as before except TEC_Idle.

File: initfiles/bin/init_thor [hunk 1]
```diff
@@ -186,13 +188,15 @@ while [[ 1 ]]; do
...
-                fi
+              log "Running autoswap $THORNAME :: ($thorpid)"
+              swapnode auto $DALISERVER $component
+              errcode=$?
+              if [[ 0 != ${errcode} ]]; then
+                log "Auto swap node failed, errcode=${errcode}"
+                killed
+              fi
+            else
+              log "Autoswap set to false.  restarting"
```
Comment (jakesmith) [inline initfiles/bin/init_thor]: trivial: double space, should it be : .... false, restarting

**Comment by jakesmith** [comment]
@Michael-Gardner - it looks fine afaics, have u tested idle timeout restart?

**Comment by Michael-Gardner** [comment]
Yeah I shortened the timer to 2 min and let it loop for a while, ran as expected.

**Comment by Michael-Gardner** [comment]
@richardkchapman should be ready for your eyes.

**Pull request merged**